packages/x-charts/src/BarChart/BarChart.tsx

```diff
@@ -25,11 +30,21 @@ export function BarChart(props: BarChartProps) {
 
   return (
     <ChartContainer
-      series={series}
+      series={series.map((s) => ({ ...s, type: 'bar' }))}
       width={width}
       height={height}
       margin={margin}
-      xAxis={xAxis}
+      xAxis={
+        xAxis ?? [
+          {
+            id: DEFAULT_X_AXIS_KEY,
+            scaleType: 'band',
+            data: [...new Array(Math.max(...series.map((s) => s.data.length)))].map(
```

Comment:
I'm not sure about this `data` defaulting. It seems somewhat strange to provide a component with such functionality (a demo Wrapper - sure, but a final component..?)
Are any other chart libraries doing anything similar? 🤔

Reply:
For recharts and nivo they don't have the problem because x and y values are in a unique object: `{ axisValue, y1, y2, ...}`

Echarts has a similar behavior to the one I implemented

This default has one main goal: avoiding the charts to fail because of missing information

Reply:
Gotcha. Well, then we can proceed with this similar approach if you feel that it would be the best one 🤔 
The other way could be to render an empty chart or nothing if the required props are not provided. After all, that's why they would be marked as required via TS and PropTypes. 🤔